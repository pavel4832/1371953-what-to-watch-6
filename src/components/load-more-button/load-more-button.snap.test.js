import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import LoadMoreButton from './load-more-button';
import configureStore from 'redux-mock-store';
import MOVIES from '../../mock/movies';

const mockStore = configureStore({});

it(`Should LoadMoreButton render correctly`, () => {
  const store = mockStore({
    DATA: {filteredMovies: MOVIES, renderedMovieCount: 8},
  });
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <LoadMoreButton />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
