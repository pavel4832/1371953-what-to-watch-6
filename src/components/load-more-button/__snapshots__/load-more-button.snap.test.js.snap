// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LoadMoreButton render correctly 1`] = `
<div>
  <div
    class="catalog__more"
  >
    <button
      class="catalog__button"
      type="button"
    >
      Show more
    </button>
  </div>
</div>
`;
